src/protocols/MeterS0.cpp

```diff
@@ -238,31 +238,34 @@ void MeterS0::counter_thread()
 		if (is_blocking) {
 			bool timeout = false;
 			if (_hwif->waitForImpulse( timeout )) {
+				//Event erkannt. Wert des Events ist, aufgrund des prellen reiner Zufall!!
+				//deswegen erst entprellen .....
```

Comment:
Hallo,
ich verstehe nicht,was ich bei ++impulses_neg ändern soll. Aus mir unerfindlichen Gründen war das hier bereits immer berücksichtigt.

`				if (_hwif_dir && ( _hwif_dir->status()>0 ) ) {
					++_impulses_neg;
				} else if ( _hwif->status()>0 ){ // Impulse nur bei echter logisch 1 werten!!
					++_impulses;
                }`

Oder meinst Du ein zweites 
`if (_hwif->waitForImpulse( timeout )) { ......}`
bei dem statt `_hwif->waitForImpulse()`  der Zeiger `_hwif_dir->waitForImpulse` verwendet wird?
Wenn Du das meinst verstehe ich allerdings nicht , was die bisherigen ++impulses_neg überhaupt in dem if verloren haben. Ich bin davon ausgegegangen das der bislang auschließlich verwendete `_hwif->waitForImpulse()` für alle Arten von GPIO aufgerufen wird. Bei Zweirichtungszählern gibt es zusätzlich den _hwif_dir Zeiger. Sollte der gesetzt sein wird der Impulse negativ gewertet, sonst nicht. Warum hier schon seit immer auch gleichzeitig der Status mit ausgewertet wurde entzieht sich meiner Kenntnis. Ich bin der Meinung, da ist nichts zu ändern.

Weiterhin habe ich bislang nur den 

> is_blocking Fall

berücksichtigt. Der 

> non blocking Fall

im nächsten Haupt If macht eventuell auch den gleichen Auswertefehler. Allerdings weiß ich im Moment noch nicht, wann dieser Fall überhaupt eintritt.